netdog: refactor to prepare for upcoming additions

zmrow opened the pull request:
**Issue number:**
Related to #2204 

**Description of changes:**
This refactor is in preparation for the upcoming additions related to static addressing.

Almost no code was changed minus the few clippy lints that are in the last commit.  The majority of the code was moved around as is.
```
    netdog: Extract lease parsing functions to module
```
```
    netdog: refactor cli code into modules
    
    Since `main.rs` was starting to get large and unwieldy, and additional
    code is expected to be added, this commit refactors all the CLI code
    into separate files grouped under a `cli` module.
```
```
    netdog: implement the suggested clippy lints
```




**Testing done:**
* Unit tests continue to pass
* Build and run an `aws-k8s-1.21` image successfully (`current_ip`, `resolv.conf`, `primary_interface`, and `/etc/sysctl.d/90-primary_interface.conf` all get properly written)
```
bash-5.1# cat /etc/sysctl.d/90-primary_interface.conf 
-net.ipv6.conf.eth0.accept_ra = 2
-net.ipv4.conf.eth0.rp_filter = 2
```
* Build and run a `metal-dev` image successfully (`net.toml` correctly parsed and all above files written)
* Manually run `netdog`'s settings generator commands and observe proper output



**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

zmrow committed:
netdog: Extract lease parsing functions to module

zmrow committed:
netdog: refactor cli code into modules

Since `main.rs` was starting to get large and unwieldy, and additional
code is expected to be added, this commit refactors all the CLI code
into separate modules grouped under a `cli` module.

zmrow committed:
netdog: implement the suggested clippy lints

zmrow commented on sources/api/netdog/src/net_config/mod.rs:
```diff
@@ -29,6 +29,7 @@ pub(crate) trait Interfaces {
     /// Converts the network config into a list of `WickedInterface` structs, suitable for writing
     /// to file
+    #[allow(clippy::wrong_self_convention)]
     fn into_wicked_interfaces(&self) -> Vec<WickedInterface>;
```
In the original PR for this code, we agreed on _not_ consuming `self`.  I don't want to change the interface as I think it's pretty clear so I'm ignoring the lint.

cbgbt replied:
I think `as_$THING` is the idiomatic naming for this sort of conversion.

zmrow replied:
^ Ahhh, yeah I like that

zmrow commented:
The above force push moves the `parse_lease_info()` function into an associated function of `LeaseInfo`; it makes more sense this way. :)

bcressey commented on sources/api/netdog/src/cli/generate_hostname.rs:
```diff
@@ -0,0 +1,32 @@
+use super::{error, print_json, Result};
+use crate::CURRENT_IP;
```
What do you think about moving the constants into some other module, like the parent?

zmrow replied:
I spent more time than I'd like to admit playing and thinking about that...  I ended up keeping them where they were since they get used in the CLI _and_ other modules so I didn't really like them in `cli/mod.rs` and I also didn't really like the idea of making a top-level `lib.rs` just for constants.

Totally willing to entertain this though - perhaps there's an elegant way I just couldn't come up with! :smile:

bcressey reviewed (approved):
Nice!

zmrow committed:
netdog: change 'into_wicked_interfaces' to 'as_...'

The `as_$THING` function signature is more idiomatic.  It also avoids a
clippy lint regarding `into_$THING` function signatures.

zmrow commented:
^ Adds a commit changing the `into_wicked_interfaces()` function signature into `as_wicked_interfaces()`... much better.
